aws-py-oidc-provider-pulumi-cloud/__main__.py

```diff
@@ -10,41 +11,15 @@
 oidc_idp_url = 'https://api.pulumi.com/oidc'
 base_url = 'api.pulumi.com/oidc'
 
-# Obtain the OIDC IdP URL and form the configuration document URL
-print("Forming configuration document URL...")
-configuration_url = f'{oidc_idp_url}/.well-known/openid-configuration'
+# Obtain the certificate of the URL provided
 
-# Locate "jwks_uri" and extract the domain name
-print("Extracting domain name from jwks_uri...")
-response = requests.get(configuration_url)
-jwks_uri = response.json().get('jwks_uri', '')
-domain_name = jwks_uri.split('/')[2]
-
-# Run OpenSSL command to get certificates
-print("Retrieving OpenSSL certificates (this will take some time)...")
-command = f'openssl s_client -servername {domain_name} -showcerts -connect {domain_name}:443'
-result = subprocess.run(command, shell=True, capture_output=True, text=True)
-certificates = result.stdout.split('-----END CERTIFICATE-----')
-
-# Get the last certificate from the output
-print("Retrieving last OpenSSL certificate...")
-last_certificate = certificates[-2] + '-----END CERTIFICATE-----'
-
-# Save the certificate to a file
-print("Saving certificate to file...")
-with open('certificate.crt', 'w') as file:
-    file.write(last_certificate)
-
-# Get the thumbprint of the final certificate
-print("Retrieving certificate thumbprint...")
-x509 = OpenSSL.crypto.load_certificate(OpenSSL.crypto.FILETYPE_PEM, last_certificate)
-thumbprint = (x509.digest('sha1').decode()).replace(":", "")
+certificate = tls.get_certificate(url=oidc_idp_url)
 
 # Create an OIDC identity provider
 print("Creating OIDC provider...")
 oidc_provider = iam.OpenIdConnectProvider("oidcProvider",
     client_id_lists=[audience],
-    thumbprint_lists=[thumbprint],
+    thumbprint_lists=[certificate.certificates[0].sha1_fingerprint],
```

Comment:
Root CA only

Reply:
If you run this:

```typescript
import * as tls from "@pulumi/tls";

export = async () => {
    const certs = tls.getCertificateOutput({
        url: "https://api.pulumi.com/oidc"
    });

    return {
        thumbprint: certs.certificates[0].sha1Fingerprint
    }
}
```

the thumbprint is the same as what's in IAM > Identity Providers > api.pulumi.com/oidc 

and look at the thumbprint value.

Reply:
`9e99a48a9960b14926bb7f3b02e22da2b0ab7280`